### ISSUE TITLE
Hover on Array[String] causes metals to log error (but all works?)

### ISSUE THREAD

**Issue by kpbochenek** [scalameta/metals#1788]
Create file:
```scala

object Main {
  def main(args: Array[String]): Unit = {
    
    println("OK!")
    //println(Helper.help())
  }

  final val x = 3
}
```

build.sbt
```
name := "aaa"
scalaVersion := "2.13.2"
```

Hover over `Array]` , logs:
something's wrong: no file:///\~/vl/ttt/src/main/scala/Main.scala in Array[String]RangePosition(file:///\~/vl/
ttt/src/main/scala/Main.scala, 32, 32, 45)

**Comment by tgodzik** [comment]
Thanks for finding this! I can confirm that it's an issue. No idea though where that warning is coming from. We need to investigate further.

**Comment by kpbochenek** [comment]
Error comes from a scala compiler:
https://github.com/scala/scala/blob/17570dfb9292a3938896935c1f941099d1e6e17f/src/interactive/scala/tools/nsc/interactive/Global.scala#L348

On metals side we run compiler for hover here:
`pc.hover(CompilerOffsetParams.fromPos(pos, token))`

And `pc` was fed with proper url and pos points correctly to `Array`

Not much we can do on metals side.

**Comment by tgodzik** [comment]
@kpbochenek could you maybe see if it's reported in Scala and Dotty? If it's not we should report it or even fix if possible.

**Comment by kpbochenek** [comment]
Will check for scala2

**Comment by kpbochenek** [comment]
I tried to reproduce it in scala, I added test and actually no error message is printed there:
https://github.com/kpbochenek/scala/pull/1/files

From metals perspective:
We call `HoverProvider.hover()` and
then in `tree match {` our Tree is `TypeApply` as we point to `Array` at this point.
Then we call `          val expanded = expandRangeToEnclosingApply(pos)`
and it expands correctly so that `case Some(value) =>` has position for `Array[String]` (31, 31, 44) - like in my scala/scala added test.
but then we call `typedTreeAt(value.pos)` and it generates error log but result is correct `scala.Array[String]`

Funny enough checking this example:
`  def xxx(args: Option[String]): Option[String] = ???`

First `Option` reports error but shows documentation
Second `Option` doesn't print error but doesn't show documentation either!

Yeah ok for a TYPE of method docs/hover is not shown, it is not related to this case.
Ok even more complicated, first time you hover on TYPE of method a popup with docs/signature is shown, every other hover shows nothing.

Gonna leave my findings here hopefully it will help fix it, I have no idea what is going on here.
